platform/contracts/admin/src/state/contracts.rs

```diff
@@ -68,3 +75,67 @@ pub(crate) fn load_all(storage: &dyn Storage) -> Result<ContractsGroupedByProtoc
             .map_err(Into::into)
     })
 }
+
+#[cfg(feature = "migrate")]
+pub(super) fn migrate_protocols(
+    storage: &mut dyn Storage,
+    api: &dyn Api,
+    querier: QuerierWrapper<'_>,
+    mut reserve_contracts: BTreeMap<String, String>,
+) -> Result<()> {
+    Map::<'_, String, OldProtocol>::new(
+        std::str::from_utf8(PROTOCOL.namespace()).expect("Expected valid UTF-8 encoded key!"),
+    )
+    .range(storage, None, None, Order::Ascending)
+    .collect::<sdk::cosmwasm_std::StdResult<Vec<_>>>()
+    .map_err(Into::into)
+    .and_then(|protocols| {
+        protocols.into_iter().try_for_each(|(name, protocol)| {
+            let Some(reserve) = reserve_contracts.remove(&name) else {
+                return Err(Error::MissingProtocol(name));
+            };
+
+            let reserve = api.addr_validate(&reserve)?;
+
+            platform::contract::validate_addr(querier, &reserve).unwrap();
```

Comment:
why `unwrap()`? It is preferable to return an error

Reply:
That one I didn't expect. I must have somehow missed it after the initial version of just unwraps. That caught me off guard for sure. Thanks for catching it!